File: daily_challenge_leetcode/1015.py
class Solution:
    def smallestRepunitDivByK(self, k: int) -> int:
        if k == 1:
            return 1
        num = 0
        for length in range(1, k+1):
            num = (num * 10 + 1) % k
            if num == 0:
                return length
        return -1

File: daily_challenge_leetcode/1015.js
/**
 * @param {number} k
 * @return {number}
 */
var smallestRepunitDivByK = function (k) {
    if (k === 1) {
        return 1
    }
    let num = 0
    for (let length = 1; length <= k; length++) {
        num = (num * 10 + 1) % k
        if (num === 0) {
            return length
        }
    }
    return -1
};